### PR TITLE
dvc: make graph consist of stages not their paths

### PR DESCRIPTION
**Pull request by Suor**

### CODE REVIEW

File: dvc/repo/__init__.py [hunk 1]
```diff
@@ -374,6 +355,7 @@ def graph(self):
     def pipelines(self):
         return get_pipelines(self.graph)
 
+    # TODO: refactor
```
Comment (shcheklein) [inline dvc/repo/__init__.py]: please, let's not put TODOs - create a ticket for this.

File: dvc/stage.py [hunk 2]
```diff
@@ -241,6 +241,20 @@ def __repr__(self):
             path=self.relpath if self.path else "No path"
         )
 
+    def __hash__(self):
+        return hash(self.path_in_repo)
+
+    def __eq__(self, other):
+        return (
+            self.__class__ == other.__class__
+            and self.repo == other.repo
```
Comment (shcheklein) [inline dvc/stage.py]: do we have a "semantic" `__eq__` for the `repo` class?

Comment (Suor) [reply dvc/stage.py]: No, I meant the same repo here. May just change it to `is` to make the intention clear.

**Comment by Suor** [comment]
Removed a TODO, changed repo comparison to `is`.

**Comment by shcheklein** [comment]
@Suor thanks 🙏

**Pull request merged**